### ISSUE TITLE
All examples in tutorial_wrapper segfault in Ubuntu 16.04

### ISSUE THREAD

**Issue by pbustos** [CMU-Perceptual-Computing-Lab/openpose#504]
### Posting rules
1. **Fill** the **Your System Configuration section (all of it!)** if you have some kind of error or performance question.
2. **No** questions about **training**. OpenPose only implements testing.
3. **No** questions about **Caffe installation errors/issues**. Check [Caffe](http://caffe.berkeleyvision.org) documentation and help for those errors.
4. **No** questions about **CUDA check failed errors**. These errors are usually fixed by re-installing CUDA, re-installing the proper cuDNN version, and re-compiling (or re-installing) OpenPose. Otherwise, check for help in CUDA forums.
5. **No duplicated** posts.
6. **No** posts about **questions already answered / clearly explained in** the **documentation** (e.g. **no more low-speed nor out-of-memory questions**).
7. Set a **proper issue title**: add the Ubuntu/Windows word and be specific (e.g. do not simple call it: `Compile error`).
8. Only English comments.
Issues/comments which do not follow these rules will be **ignored or removed** with no further clarification.

### Issue Summary
All samples in the build/examples/tutorial_wrapper/       segfault even compiled from a new cloned repo and even in one month ago versions of the repo.
Samples in tutorial_thread and openpose.bin run ok.
It looks like it might have something to do with the op::wrapper class which I am using for integration.

Thanks!

### Executed Command (if any)
Note: add `--logging_level 0 --disable_multi_thread` to get higher debug information.
robolab@robolab-muecas:~/software/openpose$ build/examples/tutorial_wrapper/2_user_synchronous.bin --logging_level 0 --disable_multi_thread

### OpenPose Output (if any)
Starting pose estimation demo.
/home/robolab/software/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToPoseModel():16
/home/robolab/software/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToScaleMode():53
/home/robolab/software/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToHeatMapScaleMode():82
/home/robolab/software/openpose/examples/tutorial_wrapper/2_user_synchronous.cpp:openPoseTutorialWrapper2():420
Configuring OpenPose wrapper. In /home/robolab/software/openpose/examples/tutorial_wrapper/2_user_synchronous.cpp:openPoseTutorialWrapper2():441
/home/robolab/software/openpose/include/openpose/wrapper/wrapper.hpp:configure():435
/home/robolab/software/openpose/src/openpose/wrapper/wrapperAuxiliary.cpp:wrapperConfigureSecurityChecks():18
/home/robolab/software/openpose/src/openpose/wrapper/wrapperAuxiliary.cpp:wrapperConfigureSecurityChecks():148
Auto-detecting all available GPUs... Detected 1 GPU(s), using 1 of them starting at GPU 0.
/home/robolab/software/openpose/include/openpose/wrapper/wrapper.hpp:configure():603
/home/robolab/software/openpose/include/openpose/wrapper/wrapper.hpp:configure():937
Starting thread(s)
/home/robolab/software/openpose/include/openpose/wrapper/wrapper.hpp:configureThreadManager():1288
/home/robolab/software/openpose/include/openpose/wrapper/wrapper.hpp:exec():951
/home/robolab/software/openpose/include/openpose/thread/threadManager.hpp:exec():163
/home/robolab/software/openpose/include/openpose/thread/queueBase.hpp:addPusher():360
/home/robolab/software/openpose/include/openpose/thread/queueBase.hpp:addPusher():360
/home/robolab/software/openpose/include/openpose/thread/queueBase.hpp:addPusher():360
/home/robolab/software/openpose/include/openpose/thread/threadManager.hpp:exec():168
/home/robolab/software/openpose/include/openpose/thread/thread.hpp:threadFunction():182
Starting initialization on thread. In /home/robolab/software/openpose/src/openpose/pose/poseExtractorCaffe.cpp:netInitializationOnThread():177
Segmentation fault

### Type of Issue
You might select multiple topics, delete the rest:
- Execution error

### Your System Configuration
**Operating system** (`lsb_release -a` in Ubuntu): 
No LSB modules are available.                                                                                                       
Distributor ID: Ubuntu                                                                                                              
Description:    Ubuntu 16.04.4 LTS                                                                                                  
Release:        16.04                                                                                                               
Codename:       xenial 

**Installation mode**: CMake
**CUDA version** (`cat /usr/local/cuda/version.txt` in most cases): CUDA Version 8.0.61
**cuDNN version**: 5.1.10
**CMake version** (`cmake --version` in Ubuntu): 3.5.1
**Release or Debug mode**? (by defualt: release): release
**3-D Reconstruction module added**? (by default: no): no
**GPU model** (`nvidia-smi` in Ubuntu): GeForce GTX 1080
**Caffe version**: Default from OpenPose
**OpenCV version**: compiled from source? If so, 3.4.1
Compiler (`gcc --version` in Ubuntu): 5.4.0

**Comment by gineshidalgo99** [comment]
I have just tried them from the latest version and they work perfectly fine.

If you are using your custom wrapper.hpp, you might have to re-create it from the latest version `wrapper.hpp`.

Ideally the code shouldn't change, and we don't change it much, but the wrapper class is a different story (it's impossible not to change the wrapper every time a new feature is added). Lastly, we have added `3d` and CPU versions (+ other tiny improvements), so it had to be adapted for all of that.
